query-service-impl/src/main/java/org/hypertrace/core/query/service/projection/ProjectionTransformation.java

```diff
@@ -169,7 +169,7 @@
       ProjectionExpression projectionExpression) {
     Single<List<Expression>> argumentListSingle =
         Observable.fromIterable(projectionExpression.getArgumentsList())
-            .flatMapSingle(this::rewriteProjectionAsQueryExpression)
```

Comment:
@aaron-steinfeld Don't you need to change on L133 too?

Reply:
Line 133 is different (hence the initial mistake). When going from a maybe -> anything or single -> anything there's no source ordering to concern, so flatmap and concatmap are equivalent (concat's impl actually just calls flat). Going from something with its own order (an observable, or flowable which we aren't using), then you have to be careful when picking flat (merge), concat (sequential), or switch (latest).